packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -790,6 +791,19 @@ function performConcurrentWorkOnRoot(root) {
     return null;
   }
 
+  // TODO: We only check `didTimeout` defensively, to account for a Scheduler
+  // bug where `shouldYield` sometimes returns `true` even if `didTimeout` is
+  // true, which leads to an infinite loop. Once the bug in Scheduler is
+  // fixed, we can remove this, since we track expiration ourselves.
```

Comment:
This comment isn't quite anymore since the particular infinite loop bug can no longer happen, though there might be a related bug. Can  ou update it to match the new reason why we have this check?

Reply:
Good point, updated!